exchanges/poloniex/poloniex_futures.go

```diff
@@ -0,0 +1,741 @@
+	}
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	if offset > 0 {
+		params.Set("from", strconv.FormatUint(offset, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var resp []*LiquidationPrice
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"liquidationOrder", params), &resp)
+}
+
+// GetFuturesMarket get the market information of trading pairs in the past 24 hours.
+func (e *Exchange) GetFuturesMarket(ctx context.Context, symbol string) ([]*FuturesTickerDetails, error) {
+	params := url.Values{}
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	var resp []*FuturesTickerDetails
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"tickers", params), &resp)
+}
+
+// GetFuturesIndexPrice get the current index price.
+func (e *Exchange) GetFuturesIndexPrice(ctx context.Context, symbol string) (*InstrumentIndexPrice, error) {
+	params := url.Values{}
+	if symbol != "" {
+		params.Set("symbol", symbol)
+	}
+	var resp *InstrumentIndexPrice
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"indexPrice", params), &resp)
+}
+
+// GetIndexPriceComponents get the index price components for a trading pair.
+func (e *Exchange) GetIndexPriceComponents(ctx context.Context, symbol string) (*IndexPriceComponent, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	var resp *IndexPriceComponent
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"indexPriceComponents", params), &resp)
+}
+
+// GetIndexPriceKlineData obtain the K-line data for the index price.
+func (e *Exchange) GetIndexPriceKlineData(ctx context.Context, symbol string, interval kline.Interval, startTime, endTime time.Time, limit uint64) ([]*FuturesIndexPriceData, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	intervalString, err := intervalToString(interval)
+	if err != nil {
+		return nil, err
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("interval", intervalString)
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if err := common.StartEndTimeCheck(startTime, endTime); err != nil {
+			return nil, err
+		}
+		params.Set("sTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+		params.Set("eTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var resp []*FuturesIndexPriceData
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"indexPriceCandlesticks", params), &resp)
+}
+
+// GetFuturesMarkPrice get the current mark price.
+func (e *Exchange) GetFuturesMarkPrice(ctx context.Context, symbol string) (*FuturesMarkPrice, error) {
```

Comment:
empty symbol returns a slice please update. 

```
Poloniex raw response: {"code":200,"msg":"Success","data":[{"s":"XRP_USDT_PERP","mPx":"2.6501"},{"s":"TRX_USDT_PERP","mPx":"0.30072"},{"s":"ETH_USDT_PERP","mPx":"4170.9"},{"s":"BTC_USDT_PERP","mPx":"114597.74"},{"s":"BCH_USDT_PERP","mPx":"557.31"},{"s":"1000PEPE_USDT_PERP","mPx":"0.0073687"},{"s":"APT_USDT_PERP","mPx":"3.587"},{"s":"AVAX_USDT_PERP","mPx":"20.841"},{"s":"SOL_USDT_PERP","mPx":"200.436"},{"s":"1000SHIB_USDT_PERP","mPx":"0.010586"},{"s":"BNB_USDT_PERP","mPx":"1139.475"},{"s":"DOGE_USDT_PERP","mPx":"0.20608"},{"s":"LTC_USDT_PERP","mPx":"100"}]}
```

error: json: cannot unmarshal array into Go struct field .data of type poloniex.FuturesMarkPrice